dlpy/applications/efficientnet.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env python
+# encoding: utf-8
+#
+# Copyright SAS Institute
+#
+#  Licensed under the Apache License, Version 2.0 (the License);
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+import os
+
+from dlpy.model import Model
+from dlpy.layers import Conv2d, BN, OutputLayer, Input, GroupConv2d, GlobalAveragePooling2D, Res, InputLayer, Reshape, Scale
+from .application_utils import get_layer_options, input_layer_options
+from dlpy.utils import DLPyError
+from dlpy.network import extract_input_layer, extract_output_layer, extract_conv_layer
+import math
+
+# default mobilenet v2 block parameters:
+_MBConv_BLOCKS_ARGS = [
+ #  (expansion, out_filters, num_blocks, kernel size, stride)
+    (1, 16, 1, 3, 1, 0.25),
+    (6, 24, 2, 3, 2, 0.25),
+    (6, 40, 2, 5, 2, 0.25),
+    (6, 80, 3, 3, 2, 0.25),
+    (6, 112, 3, 5, 1, 0.25),
+    (6, 192, 4, 5, 2, 0.25),
+    (6, 320, 1, 3, 1, 0.25)
+]
+
+def EfficientNet(conn, model_table='EfficientNet', n_classes=100, n_channels=3, width=224, height=224,
+                 width_coefficient=1, depth_coefficient=1, dropout_rate=0.2, drop_connect_rate=0, depth_divisor=8,
+                 activation_fn='relu', blocks_args=_MBConv_BLOCKS_ARGS,
+                 offsets=(255*0.406, 255*0.456, 255*0.485), norm_stds=(255*0.225, 255*0.224, 255*0.229),
+                 random_flip=None, random_crop=None, random_mutation=None):
+    '''
+    Generates a deep learning model with the EfficientNet architecture.
+    The implementation is revised based on
+    https://github.com/keras-team/keras-applications/blob/master/keras_applications/efficientnet.py
+
+    Parameters
+    ----------
+    conn : CAS
+        Specifies the CAS connection object.
+    model_table : string or dict or CAS table, optional
+        Specifies the CAS table to store the deep learning model.
+    n_classes : int, optional
+        Specifies the number of classes. If None is assigned, the model will
+        automatically detect the number of classes based on the training set.
+        Default: 1000
+    n_channels : int, optional
+        Specifies the number of the channels (i.e., depth) of the input layer.
+        Default: 3
+    width : int, optional
+        Specifies the width of the input layer.
+        Default: 224
+    height : int, optional
+        Specifies the height of the input layer.
+        Default: 224
+    width_coefficient: float,
```

Comment:
can you please add the default values here

Reply:
fixed